catalyst_voices/lib/widgets/indicators/voices_no_internet_connection_banner.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:flutter/material.dart';
+
+class NoInternetConnectionBanner extends StatelessWidget {
+  final VoidCallback onRefresh;
+
+  const NoInternetConnectionBanner({
+    super.key,
+    this.onRefresh = _noop,
+  });
+
+  static void _noop() {}
```

Comment:
Usually the approach for optional callbacks is to make them just nullable. Many built-in widgets that accept `onTap` or `onPressed` callbacks have hidden logic behind this. If the callback is null then the widget would render in a disabled state (i.e. button). Here you can make the onRefresh nullable, if it's null then hide the refresh buttons.

In case there's a use case which doesn't require the callback then simply it would be set to null.